docs/data/data-grid/filtering-recipes/QuickFilterOutsideOfGrid.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+import Box from '@mui/material/Box';
+import Grid from '@mui/material/Grid';
+import { DataGrid, GridToolbarQuickFilter, GridToolbar } from '@mui/x-data-grid';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+function MyCustomToolbar(props: any) {
+  return (
+    <React.Fragment>
+      {ReactDOM.createPortal(
+        <GridToolbarQuickFilter />,
+        document.getElementById('filter-panel')!,
+      )}
+      <GridToolbar {...props} />
+    </React.Fragment>
+  );
+}
+
+const VISIBLE_FIELDS = ['name', 'rating', 'country', 'dateCreated', 'isAdmin'];
+
+export default function QuickFilterOutsideOfGrid() {
+  const { data } = useDemoData({
+    dataSet: 'Employee',
+    rowLength: 1000,
+  });
+
+  // Otherwise filter will be applied on fields such as the hidden column id
+  const columns = React.useMemo(
+    () => data.columns.filter((column) => VISIBLE_FIELDS.includes(column.field)),
+    [data.columns],
+  );
```

Comment:
We should probably follow this guide here instead: https://mui.com/x/react-data-grid/filtering/quick-filter/#excluding-hidden-columns